src/nvim/eval.c

```diff
@@ -15140,8 +15140,11 @@ static void f_sockconnect(typval_T *argvars, typval_T *rettv, FunPtr fptr)
     return;
   }
 
+  snprintf((char *)IObuff, sizeof(IObuff), "%s:%" PRIdLINENR,
```

Comment:
Also I don't think channel is lower-lever. Rather it is "specific functionality" while "eval" and "lua" are "general purpose". `event/` is the low-lowel abstraction.